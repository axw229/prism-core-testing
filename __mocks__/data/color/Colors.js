// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
