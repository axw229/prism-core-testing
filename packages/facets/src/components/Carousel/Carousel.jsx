// @flow
import React, { useContext } from 'react'
import { useIntl } from 'react-intl'
import { Carousel as ToolkitCarousel } from '@prism/toolkit'
import ConfigurationContext, {
  type ConfigurationContextType
} from '../../contexts/ConfigurationContext/ConfigurationContext'
import Iconography from '../Iconography/Iconography'
import { withPrism } from '../ToolkitComponents'

const Carousel = (props) => {
  const { formatMessage } = useIntl()
  const { cvw = {} } = useContext<ConfigurationContextType>(ConfigurationContext)
  const { carouselBtn = {} } = cvw

  const leftButton = {
    icon: carouselBtn.iconLeft && <Iconography name={carouselBtn?.iconLeft} style={{ height: '.95rem' }} />,
    label: formatMessage({ id: 'PREVIOUS' })
  }
  const rightButton = {
    icon: carouselBtn.iconRight && <Iconography name={carouselBtn?.iconRight} style={{ height: '.95rem' }} />,
    label: formatMessage({ id: 'NEXT' })
  }

  return <ToolkitCarousel leftButton={leftButton} rightButton={rightButton} {...props} />
}

export default withPrism(Carousel)
