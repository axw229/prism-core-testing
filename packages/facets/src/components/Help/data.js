// @flow
import { shouldAllowFeature } from '../../shared/utils/featureSwitch.util'
import { FEATURE_EXCLUSIONS } from '../../constants/configurations'

export const helpTabs = [{
  'id': 0,
  'name': 'ICONS_&_BUTTONS',
  'header': 'HELPFUL_HINTS.HEADER.ICONS_&_BUTTONS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.ICONS_&_BUTTONS',
  'content': [{
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'plus-circle', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.ADD_COLOR',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.ADD_COLOR']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'folder', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.SAVE',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.SAVE']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'trash', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.TRASH_A_COLOR',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.TRASH_A_COLOR']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'clone', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.MORE_SCENES',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.MORE_SCENES']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'align-justify', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.COLOR_DETAILS',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.COLOR_DETAILS']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'signal-3', 'rotate': 235, 'size': 'lg' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.GRAB_&_REORDER',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.GRAB_&_REORDER']
  }, {
    'fontAwesomeIcon': {},
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.PAINT_SCENE',
    'iconInfoContent': []
  }],
  'imageList': '',
  'isHiddenMobile': false
}, {
  'id': 1,
  'name': 'MY_COLOR_PALETTE',
  'header': 'HELPFUL_HINTS.HEADER.MY_COLOR_PALETTE',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.MY_COLOR_PALETTE',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'myColorPaletteBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_TRASH_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.AN_ACTIVE_COLOR_STATE',
    'role': 'text',
    'imagePathKey': 'myColorPalette3'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.TAP_AND_HOLD_THE_BOTTOM_EDGE_OF_A_COLOR_SWATCH_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette4'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette5'
  }],
  'subContent': [{
    'header': 'HELPFUL_HINTS.SUB_CONTENT_HEADER.MY_COLOR_PALETTE',
    'content': 'HELPFUL_HINTS.SUB_CONTENT_DESCRIPTION.MY_COLOR_PALETTE'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_COLOR_DETAILS_ICON_TRASH_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.AN_ACTIVE_COLOR_STATE',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile3'
  }],
  'isHiddenMobile': false
}, {
  'id': 2,
  'name': 'ADDING_COLORS',
  'header': 'HELPFUL_HINTS.HEADER.ADDING_COLORS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.ADDING_COLORS',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'addColorBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_ADD_A_COLOR_ICON_IN_A_COLOR_TILE_TO',
    'role': 'text',
    'imagePathKey': 'addColor1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'addColor2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.START_A_NEW_PALETTE_TO_ADD_MORE_THAN_EIGHT_COLORS',
    'role': 'text',
    'imagePathKey': 'addColor3'
  }],
  'subContent': [{
    'header': 'HELPFUL_HINTS.SUB_CONTENT_HEADER.ADDING_COLORS',
    'content': 'HELPFUL_HINTS.SUB_CONTENT_DESCRIPTION.ADDING_COLORS'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_ADD_A_COLOR_ICON_IN_A_COLOR_TILE_TO',
    'role': 'text',
    'imagePathKey': 'addColorMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'addColorMobile2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.START_A_NEW_PALETTE_TO_ADD_MORE_THAN_EIGHT_COLORS',
    'role': 'text',
    'imagePathKey': 'addColorMobile3'
  }],
  'isHiddenMobile': false
}, {
  'id': 3,
  'name': 'COLOR_DETAILS',
  'header': 'HELPFUL_HINTS.HEADER.COLOR_DETAILS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.COLOR_DETAILS',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'colorDetailBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'colorDetail1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.PREVIEW_A_COLOR_IN_COLOR_DETAILS',
    'role': 'text',
    'imagePathKey': 'colorDetail2'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'colorDetailMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.PREVIEW_A_COLOR_IN_COLOR_DETAILS',
    'role': 'text',
    'imagePathKey': 'colorDetailMobile2'
  }],
  'isHiddenMobile': false
}, {
  'id': 4,
  'name': 'PAINTING_MY_OWN_PHOTO',
  'header': 'HELPFUL_HINTS.HEADER.PAINTING_MY_OWN_PHOTO',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.PAINTING_MY_OWN_PHOTO',
  'content': [{
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'fill-drip', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.PAINT_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.PAINT_AREA']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'brush', 'rotate': 45 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.PAINTBRUSH',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.PAINTBRUSH']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'mouse-pointer', 'rotate': -20, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.SELECT',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.SELECT']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'eraser', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.ERASE',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.ERASE']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'draw-polygon', 'rotate': 10 }, {
      'variant': 'fal',
      'icon': 'plus',
      'rotate': 0,
      'size': 'xs'
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.DEFINE_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.DEFINE_AREA']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'draw-polygon', 'rotate': 10 }, {
      'variant': 'fal',
      'icon': 'minus',
      'rotate': 0,
      'size': 'xs'
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.REMOVE_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.REMOVE_AREA']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'search-plus', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.ZOOM',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.ZOOM']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'undo-alt', 'rotate': 0 }, {
      'variant': 'fa',
      'icon': 'redo-alt',
      'rotate': 0
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.UNDO_&_REDO',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.UNDO_&_REDO'],
    'isUndoRedo': true
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'redo-alt' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.REDO',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'eye', 'flip': 'horizontal' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.HIDE_PAINT',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'info-circle' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.INFO',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'trash-alt' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.DELETE_GROUP',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'object-group' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.GROUP',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'object-ungroup' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.UNGROUP',
    'iconInfoContent': []
  }],
  'imageList': '',
  'isHiddenMobile': false
}, {
  'id': 5,
  'name': 'SAVING_MY_WORK',
  'header': 'HELPFUL_HINTS.HEADER.SAVING_MY_WORK',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.SAVING_MY_WORK',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'savingBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.NAME_YOUR_CURRENT_PHOTO',
    'role': 'text',
    'imagePathKey': 'saving1'
  }, { 'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.CLICK_SAVE_ICON_TO', 'role': 'text', 'imagePathKey': 'saving3' }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.CLICK_SAVE_ICON_TO',
    'role': 'text',
    'imagePathKey': 'savingMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.NAME_YOUR_CURRENT_PHOTO',
    'role': 'text',
    'imagePathKey': 'savingMobile2'
  }],
  'isHiddenMobile': false
}]

export const helpHeader = 'HELPFUL_HINTS.TITLE'

export const filterHelpItems = (contents: string[], contentsHiddenMobile: string[], featureExclusions: string[]): typeof helpTabs => {
  // Filtering Help Section details:
  // Icons & Buttons - Save hint
  !shouldAllowFeature(featureExclusions, FEATURE_EXCLUSIONS.documentSaving) && helpTabs[0].content.forEach((icon, index) => {
    icon.iconInfoName === 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.SAVE' && helpTabs[0].content.splice(index, 1)
  })

  const helpTabNames = helpTabs.map(item => item.name)
  contentsHiddenMobile.forEach(content => { helpTabs[helpTabNames.indexOf(content)].isHiddenMobile = true })

  return helpTabs.filter(item => contents.indexOf(item.name) !== -1)
}
