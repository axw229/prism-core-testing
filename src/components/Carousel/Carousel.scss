.prism-slick-carousel {
  &__wrapper {
   position: relative;
   width:100%;
    &__content {
      display: flex;
      justify-content: space-between;
      &__prev-btn__wrapper{
        align-items: center;
        display: flex;
        position: absolute;
        left: 0;
        top: 50%;
        transform: translate(0%, -50%);
        z-index:1000;
      }
      &__next-btn__wrapper{
        align-items: center;
        display: flex;
        position: absolute;
        right: 0;
        top: 50%;
        transform: translate(0%, -50%);
        z-index:1000;
      }
      &__buttons {
        border: 1px solid #f2f2f2;
        cursor: pointer;
        font-size: 1em;
        outline: none;
        padding: 1em;
        visibility: visible;
        &--visible {
          visibility: hidden;
        }
        &:hover {
          border: 1px solid #2cabe2;
        }
      }
      &__list__wrapper {
        height: 100%;
        padding: 3em 0;
        width: 100%;
        overflow: hidden;
      }
    }
    &__indicators {
      bottom: 0;
      display: flex;
      justify-content: center;
      margin: 10px auto;
      position: absolute;
      width: 100%;
      &__icons {
        margin: 0 5px;
        &--active {
          color: #2cabe2;
        }
        &--unactive {
          color: #c8c8c8;
        }
      }
    }
  }
}

.collection-list {
  &__wrapper {
    display: inline-flex;
    flex-wrap: wrap;
    justify-content: center;
    width: 100%;
  }
  &__container{
    white-space: nowrap;
    transition-duration: 1000ms;
    transition-timing-function: ease-in-out;
  }
}