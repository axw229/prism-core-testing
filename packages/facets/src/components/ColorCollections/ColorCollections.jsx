// @flow
import React, { useCallback, useContext, useEffect, useMemo, useState } from 'react'
import { useIntl } from 'react-intl'
import { shallowEqual, useDispatch, useSelector } from 'react-redux'
import * as GA from 'src/analytics/GoogleAnalytics'
import CardMenu from 'src/components/CardMenu/CardMenu'
import Carousel from 'src/components/Carousel/Carousel'
import ColorStripButton from 'src/components/ColorStripButton/ColorStripButton'
import CollectionDetail from 'src/components/Shared/CollectionDetail'
import ColorCollectionsTab from 'src/components/Shared/ColorCollectionsTab'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'
import ConfigurationContext, {
  type ConfigurationContextType
} from 'src/contexts/ConfigurationContext/ConfigurationContext'
import { loadCollectionSummaries } from 'src/store/actions/collectionSummaries'
import { loadColors } from 'src/store/actions/loadColors'
import './ColorCollections.scss'

export function ColorCollections() {
  const dispatch = useDispatch()
  const { formatMessage, locale } = useIntl()
  const { brandId, cvw = {} } = useContext<ConfigurationContextType>(ConfigurationContext)
  const { colorCollections = {} } = cvw
  const { pagerPosition = 'center', subtitle } = colorCollections

  const { summaries, categories } = useSelector((state) => state.collectionSummaries, shallowEqual)
  const colorMap = useSelector((state) => state.colors.items.colorMap, shallowEqual)
  useEffect(() => {
    loadCollectionSummaries(brandId, { language: locale })(dispatch)
  }, [])
  useEffect(() => {
    loadColors(brandId)(dispatch)
  }, [])

  const [tabId, setTabId] = useState()
  const [collectionData, setCollectionData] = useState([])

  useEffect(() => {
    if (typeof tabId !== 'undefined') {
      const category = categories.data.filter(({ id }) => `${id}` === `${tabId}`)[0]
      const collectionData =
        colorMap && category
          ? category.summaryIds.map((summaryId) => {
              const { colorIds, coverUrl, description, name, pdfUrl, thumbColorIds, thumbUrl } =
                summaries.data[summaries.idToIndexHash[summaryId]]
              return {
                collections: colorIds.map((id) => colorMap[id]),
                coverUrl,
                description,
                name,
                pdfUrl,
                thumbColors: thumbColorIds.map((id) => colorMap[id]),
                thumbUrl
              }
            })
          : []
      setCollectionData(collectionData)
    }
  }, [tabId, colorMap])

  useEffect(() => {
    if (categories.data.length) {
      setTabId(categories.data[0].id)
    }
  }, [categories])

  return (
    <CardMenu menuTitle={formatMessage({ id: 'COLOR_COLLECTIONS' })}>
      {(setCardShowing, setCardTitle) => (
        <>
          {subtitle && <div className='color-collections__subtitle'>{subtitle}</div>}
          <div className='color-collections__wrapper'>
            <ColorCollectionsTab collectionTabs={categories.data} showTab={setTabId} tabIdShow={tabId} />
            <div className='color-collections__collections-list' role='main'>
              <Carousel
                BaseComponent={ColorStripButtonWrapper}
                btnRefList={[]}
                data={collectionData}
                defaultItemsPerView={8}
                isInfinity={false}
                getSummaryData={(collectionSummaryData) => {
                  setCardShowing(<CollectionDetail collectionDetailData={collectionSummaryData} />)
                  setCardTitle(collectionSummaryData.name)
                  GA.event(
                    { category: 'Color Collections', action: 'Collection Click', label: collectionSummaryData.name },
                    GA_TRACKER_NAME_BRAND[brandId]
                  )
                }}
                key={tabId}
                pagerPosition={pagerPosition}
                showPageIndicators={collectionData.length > 8}
              />
            </div>
          </div>
        </>
      )}
    </CardMenu>
  )
}

const ColorStripButtonWrapper = (props: any) => {
  const { data, getSummaryData, itemNumber, btnRefList, onKeyDown } = props
  const clickHandler = useCallback(() => getSummaryData(data), [data])
  const colors = useMemo(() => data.thumbColors.slice(0, 5), [data.thumbColors])
  btnRefList[itemNumber] = React.useRef()
  let imgAltText = `${data.name}.`

  if (colors.length > 0) {
    imgAltText += ' Color group includes '
    colors.map((color, index) => {
      imgAltText += colors.length === index + 1 ? `${color.name}.` : `${color.name}, `
    })
  }

  return (
    <ColorStripButton
      bottomLabel={data.name}
      colors={colors}
      onClick={clickHandler}
      onKeyDown={onKeyDown}
      ref={btnRefList[itemNumber]}
    >
      <img className='collection__summary__top-section__image' alt={imgAltText} src={data.thumbUrl} />
    </ColorStripButton>
  )
}

export default ColorCollections
