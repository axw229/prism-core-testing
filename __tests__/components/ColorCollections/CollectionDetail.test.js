/* eslint-env jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import { CollectionDetail, collectionCover, collectionDescription, triggerPrevious, triggerNext } from 'src/components/ColorCollections/CollectionDetail'
import { getColorCollectionsData, allCollectionsData } from 'src/components/Carousel/data'
import * as Colors from '__mocks__/data/color/Colors'
import { AutoSizer } from 'react-virtualized'
import ColorWallSwatch from '../../../src/components/Facets/ColorWall/ColorWallSwatch/ColorWallSwatch';

const colors = Colors.getAllColors()
const tabIdShow = 'tab1'

let defaultProps = {
  collectionDetailData: getColorCollectionsData(colors, allCollectionsData, tabIdShow)[0],
  addToLivePalette: jest.fn()
}

const getCollectionDetail = (props) => {
  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<CollectionDetail {...newProps} />)
}

describe('CollectionDetail', () => {
  let collectionDetail
  beforeAll(() => {
    if (!collectionDetail) {
      collectionDetail = getCollectionDetail()
    }
  })

  it('should match snapshot', () => {
    expect(collectionDetail).toMatchSnapshot()
  })

  it('should render img with the src as in collectionDetailData', () => {
    expect(collectionDetail.find(`img.${collectionCover}`).exists()).toBe(true)
    expect(collectionDetail.find(`img.${collectionCover}`).prop('src')).toEqual(defaultProps.collectionDetailData.img)
  })

  it('should render collection description as in collectionDetailData', () => {
    expect(collectionDetail.find(`div.${collectionDescription}`).contains(defaultProps.collectionDetailData.name)).toBe(true)
  })

  it('should render previous trigger div', () => {
    expect(collectionDetail.find(`div.${triggerPrevious}`).exists()).toBe(true)
  })

  it('should render next trigger div', () => {
    expect(collectionDetail.find(`div.${triggerNext}`).exists()).toBe(true)
  })
})

describe('CollectionDetail with Autosizer mounted', () => {
  let collectionDetail
  let autoSizerMount
  beforeAll(() => {
    if (!collectionDetail) {
      collectionDetail = getCollectionDetail()
    }
    autoSizerMount = mount(collectionDetail.find(AutoSizer).prop('children')({ height: 200, width: 200 }))
  })

  it('should render Autosizer with ColorWallSwatch', () => {
    expect(autoSizerMount.find(ColorWallSwatch)).toHaveLength(defaultProps.collectionDetailData.collections.length)
  })

  it('should call addToLivePalette function for each ColorWallSwatch component', () => {
    autoSizerMount.find(ColorWallSwatch).map((colorWallSwatch, index) => {
      colorWallSwatch.prop('onAdd')()
      expect(defaultProps.addToLivePalette).toHaveBeenCalled()
    })
  })
})
