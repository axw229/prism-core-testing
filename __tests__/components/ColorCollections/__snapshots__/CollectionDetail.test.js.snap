// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionDetail should match snapshot 1`] = `
<div
  className="collection-detail__wrapper"
>
  <div
    className="collection-detail__info"
  >
    <img
      alt=""
      className="collection-detail__cover"
      src="https://www.sherwin-williams.com/dist/cvt/assets/images/collections/2019-shapeshifter-thumb.jpg"
    />
    <div
      className="collection-detail__description"
    />
    <a
      className="collection-detail__learn-more"
      href="https://"
    >
      LEARN MORE ABOUT THIS COLLECTION
    </a>
  </div>
  <div
    className="collection-detail__color-list"
  >
    <div
      className="collection-detail__color-list-vertical"
    >
      <AutoSizer
        disableHeight={false}
        disableWidth={false}
        onResize={[Function]}
        style={Object {}}
      >
        <Component />
      </AutoSizer>
    </div>
    <div
      className="collection-detail__vertical-controls"
    >
      <div
        className="collection-detail__trigger collection-detail__previous-trigger"
        onClick={[Function]}
        role="presentation"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fa",
              "angle-up",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
      <div
        className="collection-detail__trigger collection-detail__next-trigger"
        onClick={[Function]}
        role="presentation"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fa",
              "angle-down",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
    </div>
  </div>
</div>
`;
